WalletWasabi/Helpers/FeeHelpers.cs

```diff
@@ -0,0 +1,94 @@
+﻿using System.Collections.Generic;
+using System.Diagnostics.CodeAnalysis;
+using System.Linq;
+using NBitcoin;
+using NBitcoin.Policy;
+using WalletWasabi.Blockchain.Analysis.Clustering;
+using WalletWasabi.Blockchain.TransactionBuilding;
+using WalletWasabi.Blockchain.TransactionOutputs;
+using WalletWasabi.Exceptions;
+using WalletWasabi.Wallets;
+
+namespace WalletWasabi.Helpers;
+
+public static class FeeHelpers
+{
+	public static bool TryGetMaxFeeRateForChangeless(
+		Wallet wallet,
+		IDestination destination,
+		LabelsArray labels,
+		FeeRate startingFeeRate,
+		IEnumerable<SmartCoin> coins,
+		[NotNullWhen(true)] out FeeRate? maxFeeRate,
+		bool allowDoubleSpend = false,
+		bool tryToSign = false)
+	{
+		maxFeeRate = SeekMaxFeeRate(
+			startingFeeRate,
+			feeRate => wallet.BuildChangelessTransaction(destination, labels, feeRate, coins, allowDoubleSpend, tryToSign));
+
+		return maxFeeRate is not null;
+	}
+
+	public static bool TryGetMaxFeeRate(
+		Wallet wallet,
+		IDestination destination,
+		Money amount,
+		LabelsArray labels,
+		FeeRate startingFeeRate,
+		IEnumerable<SmartCoin> coins,
+		bool subtractFee,
+		[NotNullWhen(true)] out FeeRate? maxFeeRate,
+		bool tryToSign = false)
+	{
+		maxFeeRate = SeekMaxFeeRate(
+			startingFeeRate,
+			feeRate => wallet.BuildTransaction(destination, amount, labels, feeRate, coins, subtractFee, null, tryToSign));
+
+		return maxFeeRate is not null;
+	}
+
+	private static FeeRate? SeekMaxFeeRate(FeeRate feeRate, Action<FeeRate> buildTransaction)
```

Comment:
This function is a classic interval dichotomy. Maybe a comment indicating it would be great